index.js

```diff
@@ -55,8 +57,13 @@ HandlebarsCompiler.prototype.compile = function(data, path, callback) {
     }
 
     if (ns) {
-      key = path.replace(/\\/g,'/').replace(this.pathReplace, '').replace(/\..+?$/, '');
-      result = "if (typeof " + ns + " === 'undefined'){ " + ns + " = {} }; " + ns + "['" + key + "'] = " + source;
+      header = '';
+      if(!this.inited) {
```

Comment:
Hmm, I think you're relying on the order of compilation matching the order of concatenation, which may not be the case. May be better to try to find a way to add this to the `include` file that ends up in vendor.

Also are you counting on having this defined globally? You'll want to find a way to attach it to an existing object, like something the `handlebars` library exports.